### PR TITLE
Task 2

### LINKED ISSUE #3: Task 2
> Proszę zaimplementować produkcję (P3) i (P4) oraz napisać testy jednostkowe wraz z wizualizacją grafu powstałego z wywodu oraz umieścić produkcję oraz testy w githubie

### PR DESCRIPTION
**Pull request by Yacubane**
Closes #3

### CODE REVIEW

**Review by tomekzaw — DISMISSED** [review]
🚀

File: tests/project/test_task2.py [hunk 1]
```diff
@@ -1,16 +1,343 @@
...
+    x2, y2 = 0.5, -0.5
+    x3, y3 = -0.5, 0.5
+    x4, y4 = 0.5, 0.5
+    x5, y5 = 0.0, -0.5
+    x6, y6 = -0.5, 0.0
+    x7, y7 = 0.5, 0.0
+    x8, y8 = 0.0, 0.5
+    x9, y9 = 0.0, 0.0
+
+    x13, y13 = -0.5, 0
```
Comment (matix522) [inline tests/project/test_task2.py]: why x13, y13 and not x10, y10?

**Review by Sewerinius — DISMISSED** [review]

File: tests/project/test_task2.py [hunk 2]
```diff
@@ -54,68 +55,70 @@ def make_right_side_graph_P3():
...
+        (I10, E10), (I10, E11), (I10, E20), (I10, E21),
+        (I11, E11), (I11, E12), (I11, E21), (I11, E22),
+
+        (E00, E01), (E01, E02),
+        (E10, E11), (E11, E12),
+        (E20, E21), (E21, E22),
+
+        (E00, E10), (E10, E20),
+        (E01, E11), (E11, E21),
+        (E02, E12), (E12, E22),
```
Comment (tomekzaw) [inline tests/project/test_task2.py:137]: 22f08ce5652492095eaa9a72eace87b7

File: tests/project/test_task2.py (hunk 1 above)
Comment (Yacubane) [reply tests/project/test_task2.py]: because yes

**Review by Sewerinius — APPROVED** [review]

**Review by piaskowyk — APPROVED** [review]

File: src/GramatykiGrafoweAGH/project/utils.py [hunk 3]
```diff
@@ -8,17 +9,86 @@ def is_node_between(E1: Node, E2: Node, E3: Node) -> bool:
...
+            return res
+
+    for step, next in nexts[1]:
+        if next in visited:
+            continue
+        res = dfs(poss, next, target, path + [(step, next)], visited + [step, next], level + 1)
+        if res is not None:
+            return res
+
+    return None
```
Comment (tomekzaw) [inline src/GramatykiGrafoweAGH/project/utils.py:58]: 🤔

**Pull request merged**